WordPress/src/main/java/org/wordpress/android/ui/engagement/BottomSheetUiState.kt

```diff
@@ -1,6 +1,9 @@
 package org.wordpress.android.ui.engagement
 
+import java.io.Serializable
+
 sealed class BottomSheetUiState {
+    @Suppress("SerialVersionUIDInSerializableClass")
     data class UserProfileUiState(
```

Comment:
I'm wondering if we shouldn't mark some fields as nullable because not all are always available for all the cases. For example, if the user does not have an avatar URL, it should be `null` but not an empty string that has less straight meaning. When using the state in different places with not enough context, it looks like the state **has** to contain all the data (except for `onSiteClickListener`.)
Just leaving this note for your consideration :)

Reply:
I 100% agree with you, straight meaning is important for us to handle those fields correctly. I find many implementations use default values, but it'd be better if they are defined as nullable. 

I suspect that when handling this data with Java in the past, there was a tendency to use defensive programming to avoid crashes. However, now with Kotlin, handling nullables has become much easier. I think we can find time to refactor this.